error: ‘float fmin(float, float)’ conflicts with a previous declaration

Same error also exists for `fmax`:

```
gcc -c linuxsound.cpp  -O3 -DNDEBUG=1 -ffast-math -fexpensive-optimizations -fomit-frame-pointer -I/usr/include/SDL -D_GNU_SOURCE=1 -D_REENTRANT -fpic -Wall -fexceptions -DLINUX=1 -DSDL=1 -DVERSION_STRING=\"1.5\" -oRelease/linuxsound.o
In file included from linuxsound.cpp:41:
utility.hpp:53:35: error: ‘float fmin(float, float)’ conflicts with a previous declaration
 inline float fmin(float a, float b) { return tmin(a, b); }
                                   ^
In file included from /usr/include/c++/8.2.1/math.h:36,
                 from linuxsound.cpp:33:
/usr/include/c++/8.2.1/cmath:1438:3: note: previous declaration ‘constexpr float std::fmin(float, float)’
   fmin(float __x, float __y)
   ^~~~
In file included from linuxsound.cpp:41:
utility.hpp:65:35: error: ‘float fmax(float, float)’ conflicts with a previous declaration
 inline float fmax(float a, float b) { return tmax(a, b); }
                                   ^
In file included from /usr/include/c++/8.2.1/math.h:36,
                 from linuxsound.cpp:33:
/usr/include/c++/8.2.1/cmath:1418:3: note: previous declaration ‘constexpr float std::fmax(float, float)’
   fmax(float __x, float __y)
   ^~~~
```

--- a/makefile
+++ b/makefile
@@ -56,6 +56,6 @@ docs/doxygen/html/index.html: $(CPPSOURCES)
 ## cppcheck    : run static analysis on C++ source code
 cppcheck: $(CPPSOURCES)
 	cppcheck $(CPPSOURCES) --enable=all --platform=unix64 \
-	--std=c++11 --inline-suppr --quiet --force \
+	--std=c++03 --inline-suppr --quiet --force \
 	$(addprefix -I,$(INCLUDE_DIRS)) \
 	-I/usr/include -I/usr/include/linux

--- a/src/Makefile.macosx
+++ b/src/Makefile.macosx
@@ -37,7 +37,7 @@ SDL_MIXER_HEADERS = /Library/Frameworks/SDL_mixer.framework/Headers
 MACOSX_FRAMEWORKS = -framework Cocoa -framework OpenGL -framework SDL -framework SDL_mixer
 #-------------------------------------------------------------------------------
 
-cc = g++ -c
+cc = g++ -c -std=c++03
 debugflags = -g
 link = g++ $(debugflags) -fexceptions
 cflags = -I $(SDL_MIXER_HEADERS) $(shell sdl-config --cflags) -DLINUX=1 $(debugflags) -O3 -fomit-frame-pointer -fexceptions -ffast-math -DMACOSX

--- a/src/config.mak
+++ b/src/config.mak
@@ -110,7 +110,7 @@ else
 
 # GCC
 USE_SDL := 1
-CC := $(SOULRIDE_COMPILER)
+CC := $(SOULRIDE_COMPILER) -std=c++03
 AR := ar -r
 CC_OUT_FLAG := -o
 LIB_OUT_FLAG :=

